src/Foundatio.Extensions.Hosting/Jobs/ScheduledJobRunner.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Cronos;
+using Foundatio.Caching;
+using Foundatio.Jobs;
+using Foundatio.Lock;
+using Foundatio.Utility;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Foundatio.Extensions.Hosting.Jobs;
+
+internal class ScheduledJobRunner
+{
+    private readonly Func<IServiceProvider, IJob> _jobFactory;
+    private readonly IServiceProvider _serviceProvider;
+    private CronExpression _cronSchedule;
+    private readonly ILockProvider _lockProvider;
+    private readonly ILogger _logger;
+    private readonly DateTime _baseDate = new(2010, 1, 1);
```

Comment:
why 2010?

Reply:
I didn't change this. It's the same as it was before.